src/main/java/br/unicamp/meca/system1/codelets/ActivityTrackingCodelet.java

```diff
@@ -100,18 +100,22 @@ public void trackActionSequencePlan(ArrayList<Memory> perceptualMemories, Action
           perceptualMemories != null && 
           perceptualMemories.size() > 0) {
           ActionStep currentActionStep = actionSequencePlan.getCurrentActionStep();
-          if (currentActionStep.stopCondition(perceptualMemories)) {
-            actionSequencePlan.gotoNextAction();       
-          }        
+          ActionStep lastActionStep = actionSequencePlan.getLastActionStep();
+          if (actionSequencePlan.getCurrentActionIdIndex() == 0 || lastActionStep.needsConclusion == false) {
```

Comment:
```suggestion
          if ((actionSequencePlan.getCurrentActionIdIndex() == 0 
          || lastActionStep.needsConclusion == false)
          && (currentActionStep != null  
          && currentActionStep.executed == false 
          && currentActionStep.stopCondition(perceptualMemories)
          ) ) {
```

Suggestion to avoid these nested if blocks

Reply:
OK, I changed as the suggestion. But I usually avoid doing this, (and do the nesting ifs) to indicate that the purpose of the first if is to avoid an exception in the testing of other conditions, and the others are the real condition testing. In my point of view this makes the code clearer to the reader. I have doubts if this makes the code faster. In my opinion, this just makes the code more unreadable, as a complex test expression is of more difficult interpretation. This is not the case, in this example, though, and this is the reason I made the change.